extensions-core/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureDataSegmentConfig.java

```diff
@@ -21,12 +21,15 @@
 
 import com.fasterxml.jackson.annotation.JsonProperty;
 
+import javax.validation.constraints.NotNull;
+
 /**
- * Stores the configuration for segments written to Azure deep storage
+ * Stores the configuration for segments written to Azure deep storage.
  */
 public class AzureDataSegmentConfig
 {
   @JsonProperty
+  @NotNull
   private String container;
```

Comment:
what happens if someone wants to just load the azure extension for ingestion, will this break on start?

Reply:
I believe this config is only instantiated and validated once used (in the storage module). I tested this locally quickly by loading the azure extension, but not configuring Druid to use the extension for deep storage, and was able to step through ingestion successfully.

Reply:
ok yeah, makes sense, looking at it its only used in AzureDataSegmentKiller and AzureDataSegmentPusher so it seems fine to me